apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -472,6 +474,23 @@ export class CaseService {
       })
   }
 
+  private async deliverCaseToPolice(theCase: Case): Promise<boolean> {
+    const pdf = await getCourtRecordPdfAsString(theCase, this.formatMessage)
+    const defendantNationalIds = theCase.defendants
+      ?.filter((defendant) => !defendant.noNationalId && defendant.nationalId)
+      .map((defendant) => defendant.nationalId ?? '')
```

Comment:
Consider using `flatMap` instead to get an array with guaranteed `nationalid` values.
```suggestion
      ?.flatMap((defendant) => !defendant.noNationalId && defendant.nationalId ? [defentant.nationalId] : [])
```

Reply:
I did not know about that method. Will try it out.

Reply:
The compiler complains that `flatMap` does not exist and suggests changing the 'lib' compiler option to 'es2019' or later. Probably not something we can do here.

Reply:
I managed to simplify using `reduce` instead and the type inference is correct.